programs/bpf_loader/src/syscalls/cpi.rs

```diff
@@ -1127,29 +1324,40 @@ fn update_caller_account(
     *caller_account.lamports = callee_account.get_lamports();
```

Comment:
still half-baked thinking (near bed time!), but i think this 64 bit write can be abused similarly to https://github.com/solana-labs/solana/pull/31986#discussion_r1276382802 for `SolAccountInfo`.

That's because `ref_to_len_in_vm` from `SolAccountInfo` is free-standing unlike `AccountInfo`; no need to be adjacent to `data_addr`.

so, attacker can pass crafted `ref_to_len_in_vm` which is actually aliased to one of correct `.lamports` address. then, self-cpi and fiddle the account balance a bit so that it will cause any unintended `prev_len` mutation inside `update_caller_account`.